src/Orleans/Providers/ClientProviderRuntime.cs

```diff
@@ -170,6 +173,8 @@ public IStreamPubSub PubSub(StreamPubSubType pubSubType)
                     return grainBasedPubSub;
                 case StreamPubSubType.ImplicitOnly:
                     return implictPubSub;
+                case StreamPubSubType.MemoryBaseOnly:
+                    return memoryPubSub;
```

Comment:
Other pubsub systems are global, memory is per stream provider, and silo only, so:
- Client should throw error if memory pub sub is requested
- Silo should create new memory pubsub each request.